rfcs/convergence/props-consistency-check.md

```diff
@@ -0,0 +1,391 @@
+  children: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+AccordianItemSlots, AccordianPanelSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuSlots
+
+```ts
+(empty type)
+```
+
+MenuDividerSlots, MenuGroupSlots, MenuGroupHeaderSlots, MenuListSlots, MenuPopoverSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+```
+
+MenuItemSlots
+
+```ts
+root: ObjectShorthandProps<React.HTMLAttributes<HTMLElement>, HTMLElement>;
+icon?: React.HTMLAttributes<HTMLElement>;
+checkmark?: React.HTMLAttributes<HTMLElement>;
+submenuIndicator?: React.HTMLAttributes<HTMLElement>;
+content: React.HTMLAttributes<HTMLElement>;
+secondaryContent?: React.HTMLAttributes<HTMLElement>;
+```
+
+## Compat and HTML attributes
+
+> The components not yet using slots extend ComponentPropsCompat and provide HTML attributes by extending the associated React HTML attribute types (e.g `React.HTMLAttributes<HTMLElement>`).
+> PortalProps and PopoverTriggerProps do not extend any HTML attributes, but provide children as a property directly in their types.
+
+AvatarProps,BadgeProps,DividerProps, TooltipProps
+
+```ts
+extends ComponentPropsCompat, React.HTMLAttributes<HTMLElement>
+```
+
+ButtonProps
+
+```ts
+ButtonProps = ComponentPropsCompat & React.ButtonHTMLAttributes<HTMLElement> & {...}
+```
+
+MenuButtonProps/MenuTriggerChildProps
+
+```ts
+MenuButtonProps = Omit<ButtonProps, 'iconPosition'>  & Partial<MenuTriggerChildProps> & {}
+
+MenuTriggerChildProps extends Required<Pick<React.HTMLAttributes<HTMLElement>,
+      | 'onClick'
+      | 'onMouseEnter'
+      | 'onMouseLeave'
+      | 'onContextMenu'
+      | 'onKeyDown'
+      | 'aria-haspopup'
+      | 'aria-expanded'
+      | 'id'
+    >
+```
+
+ImageProps
+
+```ts
+extends ComponentPropsCompat,
+React.ImgHTMLAttributes<HTMLImageElement>
+```
+
+LabelProps
+
+```ts
+extends ComponentPropsCompat,
+React.LabelHTMLAttributes<HTMLLabelElement>
+```
+
+LinkProps
+
+```ts
+LinkProps = ComponentPropsCompat &
+  React.AnchorHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement> &
+  Omit<React.ButtonHTMLAttributes<HTMLAnchorElement | HTMLButtonElement | HTMLElement>, 'type'> & {...}
+```
+
+PopoverProps
+
+```ts
+// PortalProps is an interface with children and mountNode.
+// It does not provide HTML attributes
+extends Pick<PortalProps, 'mountNode'>
+```
+
+PopoverSurfaceProps
+
+```ts
+extends ComponentPropsCompat,
+React.HTMLAttributes<HTMLElement>
+```
+
+## Property overrides of native properties
+
+> Several components declare properties that are also declared by extending from the DOM types (e.g. React.HtmlAttributes<T>).
+
+checked, children, color, defaultChecked, disabled, href, image, name, onClick, open, ref, target, type, value
+
+> Several components use declar properties that are also HTML element names.
```

Comment:
from @ecraig12345 in HackMD:
Are you aware of any cases where this would cause conflicts?

Reply:
I don't think it would cause a compile or runtime issue, but it could confuse users who might think it has to be set to an HTML element.